### PR TITLE
DokuWiki with CloudFront CDN option.

### PR DESCRIPTION
**Pull request by nfriedli**
Proof of concept...

### CODE REVIEW

**Comment by selfthinker** [comment]
Can you get this from a proof of concept stage to a production-ready stage? Or did you post this here just for discussion to get some feedback first?

**Comment by nfriedli** [comment]
I dont't knew where I can obtain some feedback. So I sent a pull-questions :( Probably a really bad idea...

**Comment by selfthinker** [comment]
No, it's generally a good idea. It just should have been mentioned in the description.

**Comment by selfthinker** [comment]
I generally would like to have something like this implemented. I suggested it in the old bug tracker a few years ago: https://bugs.dokuwiki.org/index.php?do=details&task_id=2592

I would not mention CloudFront specifically because there are many other CDNs out there.
The biggest thing that's missing here is a means to *upload* all the CSS and JS to whatever CDN.

I wonder if this functionality is better suited in a plugin. Although some changes in the core are obviously necessary as well to make that possible.

File: inc/common.php [hunk 1]
```diff
@@ -565,10 +565,15 @@ function ml($id = '', $more = '', $direct = true, $sep = '&amp;', $abs = false)
         $more = str_replace(',', $sep, $more);
     }
 
-    if($abs) {
-        $xlink = DOKU_URL;
-    } else {
-        $xlink = DOKU_BASE;
+    if ($conf[cdn] && $direct) {
```
Comment (selfthinker) [inline inc/common.php]: This doesn't throw an error?
Also, the `cdn` option is missing in the configuration file.

Comment (nfriedli) [inline inc/common.php]: Splitbrain said "a patch", so I interpreted that as a core patch and not a plugin. Cf. https://github.com/splitbrain/dokuwiki/issues/1020#issuecomment-73473979

**Comment by nfriedli** [comment]
Not a good idea.
Bye.

**Comment by selfthinker** [comment]
Why did you close this?

**Comment by nfriedli** [comment]
As I said: it was not a good idea. Bye :(

**Pull request closed without merging**